File: 0791-custom-sort-string/0791-custom-sort-string.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        map<char,int>ump;
        for(char&x:s)
            ++ump[x];
        vector<int>v(26,0);
        string ans ="";
        for(char &x:order){
            if(ump.find(x)!=ump.end()){
                int freq = ump[x];
                while(ump[x]){
                    ans+=x;
                    --ump[x];
                }
                ump.erase(x);
            }
        }
        for(auto &[x,y]:ump){
            while(y--){
                ans+=x;
            }
        }
        return ans;
    }
};

File: 0791-custom-sort-string/0791-custom-sort-string.py
class Solution:
    def customSortString(self, order: str, s: str) -> str:
        hashset = {}
        for x in s:
            if x in hashset:
                hashset[x]+=1
            else :
                hashset[x] =1
        ans = ""
        for x in order :
            if x in hashset:
                ans += x*hashset[x]
                hashset.pop(x)
        for x in hashset:
            ans +=x*hashset[x]
        return ans